### DISCUSSION TITLE
What is in the EveronLife Framework?

### DISCUSSION THREAD

**Discussion by thebonbon** [EveronLife/EveronLife#145] [General]
As answered in https://github.com/EveronLife/EveronLife/discussions/15 EvernLife is a Framework. But we never set what is in this Framework.
This discussion is kinda the same as https://github.com/EveronLife/EveronLife/discussions/144 but I wanted to discuss this separately

So what should the Framework be?

> - A fully functional server?

Download the mod on a Server and you have a fully playable Server with money, persistence and a fully edited map full with shops, traders and jobs?

> - A semi functional server?

Download the mod on a Server and you have a kinda playable Server with really basic money, persistence, no map just single shop, trader and job prefabs? That you then need to use to make your own map?

> - Just really basic Systems / Scripts?

System / Scripts that you need to expand and make your own systems?

> - Really basic server framework (core)

A "core" mod for the server. You will need to add other mods that use the core mod to add Jobs, Shops and Traders etc.. (see #144 )

...
A mix of these or something else?

**Comment by Arkensor** [comment]
Something between these two.
> Download the mod on a Server and you have a fully playable Server with money, persistence and a fully edited map full with shops, traders and jobs?

> Download the mod on a Server and you have a kinda playable Server with really basic money, persistence, no map just single shop, trader and job prefabs? That you then need to use to make your own map?

I would like it to be just like altis life. You download the vanilla setup and the basic playable gameplay loop. e.g. 3 gathering resources, 3 mining, 1-2 shops in each major city + their map icons. But nothing too outstanding. They serve more as playtesting ground and basic setup that a server owner can copy or subscene if they wish and then go about customizing everything with 20 more resources or whatever.

A server owner must not spend a whole day before he is able to spawn, farm some items, buy some stuff, etc. 

And I think this was the vision at kick-off as well.

**Reply by thebonbon** [reply]
And the rest of the features like logging, farming, repair, crafting, license plates, etc.. will be a sperate mods that server owners can add?

**Reply by Arkensor** [reply]
No, they all go in. Those are all basic features of the gamemode. The vision is not a barebones technical framework, but a "base game mode". What is part of the base and what is not is feature voted by the community and managed by maintainers.

**Reply by thebonbon** [reply]
So we add them but not in the main world? Just as a prefab that server owners can then add if they want?

**Reply by Arkensor** [reply]
No no, we add each of them, the main world is just not being taken care of atm while we are prototyping everything in the debug world. If for example your logging is done we could think of a well-suited place or maybe 2 where people can do this activity (preferably near the spawn city, unless it's possible with every tree like gathering apples) so they can experience one thing of each gameplay feature in our vanilla mode. We must not spend weeks balancing everything, a basic balancing is doing alright. For "real" long-term servers they will do their own. We just do things like 1 apple is worth less than 1 gold bar etc.

**Reply by thebonbon** [reply]
So for resources we add one of each to the world and the same for shops, traders and processors.

But not really for a playable Server just for showing how they work.
Should everything really be near the city? Stuff like the saw mill is a fix place in the world that we should use.

**Reply by Arkensor** [reply]
Ah sorry, my previous message can be misunderstood. No, I do not want to condense down everything near the city. This was just specific to logging as trees are everywhere and it looks like a nice early to mid-game activity. The whole map is to be used, with fitting locations roleplaywise. We design a playable server, that might have not the best balancing or the most interesting skins and only those few vehicles AR has, but the "features" of the game mode are all in. And then its more a matter of how server owners customize. Maybe they sub-scene to only add additional things, maybe they dislike our "vanilla" map and redo it from "scratch" using the same prefabs we placed.

**Reply by thebonbon** [reply]
So it is a fully playble server and not a playtesting ground?

> Download the mod on a Server and you have a fully playable Server with money, persistence and a fully edited map full with shops, traders and jobs?

**Reply by Arkensor** [reply]
Fully playable, balancing might not be the best and very limited in terms of assets yes.
